### ISSUE TITLE
argocd app sync --local has a 2mb 

### ISSUE THREAD

**Issue by mschulman** [argoproj/argo-cd#19121]
Checklist:

* [ x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [ x] I've included steps to reproduce the bug.
* [x ] I've pasted the output of `argocd version`.

**Describe the bug**
When using app sync --local there seems to be a 2mb limit on sending the manifest/files.  It doesn't seem to obey the ARGOCD_GRPC_MAX_SIZE_MB value.  

**To Reproduce**
I'm trying to release the kube-prometheus-stack helm chart but in my case I have an helm chart in my repo that just depends on this.  

Chart.yaml : 
...
dependencies:

  - name: kube-prometheus-stack
    version: "61.3.2"
    repository: https://prometheus-community.github.io/helm-charts

setup some simple values.yaml and create an app (lets call it monitoring) and have it point to this helm chart in the repo

Syncing the app in the UI works fine with server-side set.  

But when you want to try to do a local dry run

argocd app sync monitoring --dry-run --local ./monitoring --server-side 

I get

ERRO[0004] finished unary call with code Internal        error="rpc error: code = Internal desc = error setting app operation: error updating application \"monitoring\": rpc error: code = ResourceExhausted desc = trying to send message larger than max (2878860 vs. 2097152)" grpc.code=Internal grpc.method=Sync grpc.service=application.ApplicationService grpc.start_time="2024-07-19T15:16:30-04:00" grpc.time_ms=1033.564 span.kind=server system=grpc
FATA[0004] rpc error: code = Internal desc = error setting app operation: error updating application "monitoring": rpc error: code = ResourceExhausted desc = trying to send message larger than max (2878860 vs. 2097152)

FWIW,
helm template --debug monitoring |wc
 6385   22573  639976

**Expected behavior**

Ideally we could do local dry runs as part of our CI approach.  

Not sure if I'm doing something dense, but can get this to work just fine on smaller apps/charts.  

Thanks so much

**Screenshots**

**Version**

argocd: v2.11.3
  BuildDate: unknown
  GitCommit: v2.11.3
  GitTreeState: clean
  GitTag: v2.11.3
  GoVersion: go1.22.3
  Compiler: gc
  Platform: darwin/arm64
argocd-server: v2.11.3
  BuildDate: unknown
  GitCommit: v2.11.3
  GitTreeState: clean
  GitTag: v2.11.3
  GoVersion: go1.22.3
  Compiler: gc
  Platform: darwin/arm64
  Kustomize Version: could not get kustomize version: exec: "kustomize": executable file not found in $PATH
  Helm Version: v3.15.2+gv3.15.2
  Kubectl Version: v0.24.2
  Jsonnet Version: v0.20.0
**Logs**

**Comment by juwon8891** [comment]
@mschulman 
I'm interested in this issue. Can I try to solve it?

**Comment by mschulman** [comment]
Please!  Let me know what I can do!

thanks so much...

**Comment by juwon8891** [comment]
This issue works fine in gui but can only be reproduced in cli, right?

**Comment by mschulman** [comment]
Yup.  I think the problem is in the --local part for what I assume is sending the manifests to the server.  Running it with the cli without --local [dir] also works fine.

**Comment by juwon8891** [comment]
Instead of the --local option, the --local-repo-root option is effective.
`argocd app sync monitoring --local-repo-root ./monitoring --server-side --dry-run`
[https://argo-cd.readthedocs.io/en/stable/user-guide/commands/argocd_app_sync/](url)

**Comment by olivergondza** [comment]
Closing as outdated and likely not an Argo CD bug. Please file a new issue against recent Argo CD version if still relevant.
